From 41d32b16473f81ec94cd51ed51fdc6f54e1f76d6 Mon Sep 17 00:00:00 2001
From: Thibaut Girka <thib@sitedethib.com>
Date: Fri, 25 Jun 2010 20:17:18 +0200
Subject: [PATCH] glamo-core: initialize engine states as disabled

This should fix random init failures such as white screen on boot.
---
 drivers/mfd/glamo/glamo-core.c |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/drivers/mfd/glamo/glamo-core.c b/drivers/mfd/glamo/glamo-core.c
index 33a7a59..f6674f4 100644
--- a/drivers/mfd/glamo/glamo-core.c
+++ b/drivers/mfd/glamo/glamo-core.c
@@ -916,7 +916,7 @@ static int __devinit glamo_supported(struct glamo_core *glamo)
 
 static int __devinit glamo_probe(struct platform_device *pdev)
 {
-	int ret = 0, irq, irq_base;
+	int ret = 0, n, irq, irq_base;
 	struct glamo_core *glamo;
 	struct resource *mem;
 
@@ -924,6 +924,9 @@ static int __devinit glamo_probe(struct platform_device *pdev)
 	if (!glamo)
 		return -ENOMEM;
 
+	for (n = 0; n < __NUM_GLAMO_ENGINES; n++)
+		glamo->engine_state[n] = GLAMO_ENGINE_DISABLED;
+
 	spin_lock_init(&glamo->lock);
 
 	glamo->pdev = pdev;
-- 
1.7.1

